stat/testuv/norm.go

```diff
@@ -0,0 +1,77 @@
+package testuv
+
+import (
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/stat"
+)
+
+// NormalSkewTest test whether the skew is different from the normal distribution.
+// This function tests the null hypothesis that the skewness of population that the
+// sample was drawn from is the same as that of a corresponding normal distribution.
+// The length of a must be at least 8.
+func NormalSkewTest(a []float64) float64 {
+	// Implementation based on scipy's: https://github.com/scipy/scipy/blob/v1.3.0/scipy/stats/stats.py#L1481-L1549
+	b2 := stat.Skew(a, nil)
+	n := len(a)
+	if n < 8 {
+		panic(fmt.Sprintf("testuv: skewtest is not valid with less than 8 samples, %d samples were given", n))
+	}
+	nf := float64(n)
+	y := b2 * math.Sqrt(((nf+1)*(nf+3))/(6*(nf-2)))
+	beta2 := (3 * (math.Pow(nf, 2) + 27*nf - 70) * (nf + 1) * (nf + 3) / ((nf - 2) * (nf + 5) * (nf + 7) * (nf + 9)))
+	w2 := -1 + math.Sqrt(2*(beta2-1))
+	delta := 1 / math.Sqrt(0.5*math.Log(w2))
+	alpha := math.Sqrt(2 / (w2 - 1))
+	if y == 0 {
+		y = 1
+	}
+	return delta * math.Log(y/alpha+math.Sqrt(math.Pow(y/alpha, 2)+1))
+}
+
+// NormalKurtosisTest test whether a dataset has normal kurtosis.
```

Comment:
```
// NormalKurtosis tests whether a dataset has normal kurtosis. It returns the
// 2-sided p-value for the hypothesis test.
//
// This function tests the null hypothesis that the kurtosis of the population from
// which the sample was drawn is that of the normal distribution:
//  kurtosis = 3(n-1)/(n+1).
//
// The length of a must be at least 5 otherwise NormalKurtosis will panic.
```